File: Baekjoon/S2/C#/S2_알고리즘 수업 - 깊이 우선 탐색 1.cs
https://www.acmicpc.net/problem/24479
//메모리 초과
using System;
using System.Text;

namespace Beakjoon
{
    class Program
    {
        static int[] ans;
        static int[,] map;
        static bool[] visit;
        static int n, cnt = 1;

        static void Main(string[] args)
        {
            int[] nmr = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
            n = nmr[0];
            int m = nmr[1];
            int r = nmr[2];

            map = new int[n + 1, n + 1];
            visit = new bool[n + 1];
            ans = new int[n + 1];

            for (int i = 0; i < m; i++)
            {
                int[] uv = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
                map[uv[0], uv[1]] = map[uv[1], uv[0]] = 1;
            }

            DFS(r);

            StringBuilder sb = new StringBuilder();
            for (int i = 1; i <= n; i++)
                sb.AppendLine($"{ans[i]}");

            Console.Write(sb);
        }

        static void DFS(int r)
        {
            visit[r] = true;
            ans[r] = cnt++;

            for (int i = 1; i <= n; i++)
            {
                if (!visit[i] && map[r, i] == 1)
                {
                    DFS(i);
                }
            }
        }
    }
}

File: Baekjoon/S2/JAVA/S2_알고리즘 수업 - 깊이 우선 탐색 1.java
//https://www.acmicpc.net/problem/24479

import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class Main{
    static List<List<Integer>> li;
    static int[] check;
    static int cnt = 1;
    
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int R = Integer.parseInt(st.nextToken());
        
        li = new ArrayList<>();
        for(int i = 0; i <= N; i++)
            li.add(new ArrayList<Integer>());
        
        for(int i = 0; i < M; i++){
            st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());
            
            li.get(u).add(v);
            li.get(v).add(u);
        }
        
        for(int i = 1; i <= N; i++)
			Collections.sort(li.get(i));
		
		check = new int[N+1];
        DFS(R);
        
        StringBuilder sb = new StringBuilder();
        for(int i = 1; i < check.length; i++)
			sb.append(check[i]).append("\n");
		
        System.out.print(sb);
	}
	
	static void DFS(int r){
	    check[r] = cnt;
	    
	    for(int i = 0; i < li.get(r).size(); i++){
			int tmp = li.get(r).get(i);
			if(check[tmp] == 0){
				cnt++;
				DFS(tmp);
			}
		}
	}
}